File: sdk/src/main/java/com/qiyei/sdk/base/BaseActivity.java
package com.qiyei.sdk.base;


import android.content.Context;
import android.content.Intent;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;


import com.qiyei.sdk.ioc.ViewUtils;


/**
 * @author Created by qiyei2015 on 2017/5/6.
 * @version: 1.0
 * @email: 1273482124@qq.com
 * @description: 整合应用的BaseActivity.只能放一些通用的方法，基本每个Activity都需要使用的方法，readDataBase最好不要放进来 ，
 *  如果是两个或两个以上的地方要使用,最好写一个工具类
 */
public abstract class BaseActivity extends AppCompatActivity implements View.OnClickListener {

    protected final String TAG = BaseActivity.class.getSimpleName();

    /**
     * context
     */
    protected Context mContext;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initContentView();
        ViewUtils.inject(this);
        //initData();
        //initView();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //ViewUtils.unInject(this);
    }

    /**
     * 设置setContentView布局
     */
    protected abstract void initContentView();

    /**
     * 初始化View
     */
    protected abstract void initView();
    /**
     * 初始化数据
     */
    protected abstract void initData();

    /**
     * 根据id返回对应的View对象
     * @param id
     * @param <T>
     * @return
     */
    protected  <T extends View> T ViewById(int id){
        return (T) findViewById(id);
    }

    /**
     * 启动Activity
     * @param clazz
     */
    protected void startActivity(Class<?> clazz){
        startActivity(new Intent(this,clazz));
    }

}

File: framework/Framework/src/main/java/com/qiyei/framework/ui/activity/BaseActivity.kt
package com.qiyei.framework.ui.activity



import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.FrameLayout
import com.alibaba.android.arouter.launcher.ARouter
import com.qiyei.framework.AppManager
import com.qiyei.framework.titlebar.CommonTitleBar
import com.trello.rxlifecycle3.components.support.RxAppCompatActivity

import org.jetbrains.anko.find
import org.jetbrains.anko.toast

/**
 * @author Created by qiyei2015 on 2018/9/22.
 * @version: 1.0
 * @email: 1273482124@qq.com
 * @description: 所有Activity基类
 */
abstract class BaseActivity : RxAppCompatActivity(), View.OnClickListener{

    private var preTime:Long = 0

    /**
     * 通用标题栏
     */
    protected var mTitleBar: CommonTitleBar? = null
    /**
     * Context
     */
    protected lateinit var mContext:Context
    /**
     * AppContext
     */
    protected lateinit var mAppContext:Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this
        mAppContext = applicationContext
        ARouter.getInstance().inject(this)
        AppManager.instance.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.instance.removeActivity(this)
    }

    override fun onClick(view: View) {

    }

    /**
     * 用于日志调试TAG
     */
    open fun getTAG():String{
       return this::class.java.simpleName
    }


//    override fun onBackPressed() {
//        val time = System.currentTimeMillis()
//        if (time - preTime > 2000){
//            preTime = time
//            toast("再按一次退出")
//        }else {
//            AppManager.instance.exitApplication(this)
//        }
//    }

    //获取Window中视图content
    val contentView:View
        get() {
            val content = find<FrameLayout>(android.R.id.content)
            return content.getChildAt(0)
        }
}